File: Data Structures & Algorithms/find-duplicate-integer/submission-5.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        res = 0
        slow = 0
        fast = 0
        while True:
            res = slow
            slow = nums[slow]
            fast = nums[nums[fast]]
            if slow == fast:
                return nums[res]

File: Data Structures & Algorithms/is-anagram/submission-5.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) return false;
        vector<int> fq(26, 0);
        for (int i{}; i < s.size(); ++i) {
            fq[s[i] - 'a']++;
            fq[t[i] - 'a']--;
        }
        for (int i{}; i < 26; ++i) {
            if (fq[i]) return false;
        }
        return true;
    }
};
